pkg/image/api/helper.go

```diff
@@ -843,3 +843,40 @@ func SetContainerImageEntrypointAnnotation(annotations map[string]string, contai
 	s, _ := json.Marshal(cmd)
 	annotations[key] = string(s)
 }
+
+// ImageHasSignature returns true if the signature identified by type and blob is already stored in the image.
+func ImageHasSignature(image *Image, t ImageSignatureType, b string) bool {
+	for _, signature := range image.Signatures {
+		if signature.Type == t && signature.Blob == b {
+			return true
+		}
+	}
+	return false
+}
+
+// AddImageSignature appends given signature to the image if it doesn't already have it.
+func AddImageSignature(image *Image, s *ImageSignature) {
+	if !ImageHasSignature(image, s.Type, s.Blob) {
+		if s.Created.IsZero() {
```

Comment:
is Created supposed to be a trusted field? if so, it should only be set by the system